plugin-modernizer-core/src/main/java/io/jenkins/tools/pluginmodernizer/core/impl/PluginModernizer.java

```diff
@@ -31,8 +28,11 @@ public class PluginModernizer {
 
     private static final ObjectMapper objectMapper = new ObjectMapper();
 
+    private final GoalsListCreator goalsListCreator;
+
     public PluginModernizer(Config config) {
         this.config = config;
+        this.goalsListCreator = new GoalsListCreator(config);
```

Comment:
Why not just something like creating a class responsible of maven invocation ?

I would expect to see a `start()` method of the `PluginModernizer` very simple where all the logic is encapsulated

Just an example of pseudo like (without any optimization or threading nor error handling)

```java
public void start() {
     for (String plugin : config.getPlugins()) {
            mavenInvoker.fetchMetadata(plugin)
            mavenInvoker.invokeGoal('clean')
            mavenInvoker.invokeRewrite()
     }
}
```

What do you think about trying to extract responsibility on different classes ?